dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java

```diff
@@ -366,6 +373,75 @@ public ItemRest importItem(HttpServletRequest request) throws SQLException, Auth
         return itemRest;
     }
 
+    /**
+     * Endpoint for importing item's mapped collection.
+     * The mapping for requested endpoint, for example
+     * <pre>
+     * {@code
+     * https://<dspace.server.url>/api/clarin/import/item/{ITEM_UUID}/mappedCollections
+     * }
+     * </pre>
+     * @param request request
+     * @throws SQLException       if database error
+     * @throws AuthorizeException if authorization error
+     */
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @RequestMapping(method = RequestMethod.POST, value = "/item" + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID +
+            "/mappedCollections")
+    public void importItemCollections(@PathVariable UUID uuid, HttpServletRequest request) throws SQLException,
+            AuthorizeException, IOException {
+        Context context = ContextUtil.obtainContext(request);
+        if (Objects.isNull(context)) {
+            throw new RuntimeException("Context is null - cannot import item's mapped collections.");
+        }
+
+        // Load List of collection self links
+        List<String> requestAsStringList = new ArrayList<>();
+        JSONParser parser = new JSONParser();
+        try {
+            Object obj = parser.parse(new InputStreamReader(request.getInputStream()));
+
+            for (Object entity : (JSONArray) obj) {
+                String collectionSelfLink = entity.toString();
+                requestAsStringList.add(collectionSelfLink);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot import item's mapped collections because parsing of the request JSON" +
+                    "throws this error: " + e.getMessage());
+        }
+
+        // Find Collections following its self link
+        List<DSpaceObject> listDsoFoundInRequest
+                = utils.constructDSpaceObjectList(context, requestAsStringList);
+
+        if (listDsoFoundInRequest.size() < 1) {
+            throw new UnprocessableEntityException("Not a valid collection uuid.");
+        }
+
+        for (DSpaceObject dso : listDsoFoundInRequest) {
+
+            Item item = itemService.find(context, uuid);
+            if (dso != null && dso.getType() == COLLECTION && item != null) {
```

Comment:
use Object.isNull() and equal

Reply:
That code is copied, I do not want to touch it